### PR TITLE
Adding dump() function for cfscript language JavaScript

### PR DESCRIPTION
**Pull request by MFernstrom**
I added a dump method after seeing Matthews open issue #23 

Use like so `dump(variable)`

Has the same effect as writedump(var) for complex data, has the same output as print() for simple values.

### CODE REVIEW

File: src/org/alanwilliamson/lang/javascript/convert.java [hunk 1]
```diff
@@ -126,6 +130,27 @@ public static Object print( Context cx, Scriptable thisObj, Object[] args, Funct
...
+	 */
+	public static Object dump( Context cx, Scriptable thisObj, Object[] args, Function funObj) throws Exception{
+		cfSession thisSession = (cfSession)cx.getThreadLocal("cfsession");
+		
+		try {
+			cfData dat = convert.jsConvert2cfData( (IdScriptableObject) args[0] );
+			String out = cfDUMP.dumpSession(thisSession, dat, null, "", 9999, false, false );
+			thisSession.forceWrite( out );
+			
+		} catch( Exception e ) {
```
Comment (wua22) [inline src/org/alanwilliamson/lang/javascript/convert.java]: Why catch the Exception here?

Comment (MFernstrom) [reply src/org/alanwilliamson/lang/javascript/convert.java]: Can't cast primitives/simple values to IdScriptableObject, and I couldn't figure out a decent way to test for simple/complex values  :/

I tried using cfData.isSimpleValue() but couldn't get it working going from Object to cfData if I remember right.

Comment (wua22) [reply src/org/alanwilliamson/lang/javascript/convert.java]: Ah ok, so it's a ClassCastException you're expecting in that instance.

Use "if ( args[0] instanceof IdScriptableObject )" instead.

**Review by wua22 — APPROVED** [review]

File: src/org/alanwilliamson/lang/javascript/convert.java (hunk 1 above)
Comment (MFernstrom) [reply src/org/alanwilliamson/lang/javascript/convert.java]: Aha, I didn't know that was a thing I could check for, thanks Andy!

**Pull request merged**